# Introduction To Probability
## Challenge 1

A and B are events of a probability space with $(\omega, \sigma, P)$ such that $P(A) = 0.3$, $P(B) = 0.6$ and $P(A \cap B) = 0.1$

Which of the following statements are false?
1. $ P(A \cup B) = 0.6 $
2. $ P(A \cap B^{C}) = 0.2 $
3. $ P(A \cap (B \cup B^{C})) = 0.4 $
4. $ P(A^{C} \cap B^{C}) = 0.3 $
5. $ P((A \cap B)^{C}) = 0.9 $

Since 𝑃(𝐴∩𝐵)=0.1, which is different from P(A)=0.3*P(B)=0.6 = 0.18, this means that A and B are non-mutually exclusive events.
Therefore, the following statements are false:
* Statement 1: $ [P(A \cup B) = 0.6] $ $\neq$ $[P(A)+P(B)- P(A \cap B) = 0.3+0.6-0.1=0.8]$
* Statement 2: $ P(A \cap B^{C})$ should equal $P(A \cup B)$. This is not the case, since the former is 0.2 while the latter is 0.8
* Statement 3: $ P(A \cap (B \cup B^{C})) $ = $P(A \cap B) \cup P(A \cap B^{C})$ = $ P(A \cap B) \cup P(A \cup B) $ = 0.1 + 0.8 = 0.9
* Statement 4: $ P(A^{C} \cap B^{C})$ = $ P((A \cup B)^{C})$ = $1 - P(A \cup B)$ = 1 - 0.8 = 0.2

## Challenge 2
There is a box with 10 white balls, 12 red balls and 8 black balls. Calculate the probability of:
* Taking a white ball out.
* Taking a white ball out after taking a black ball out.
* Taking a red ball out after taking a black and a red ball out.
* Taking a red ball out after taking a black and a red ball out with reposition.

**Hint**: Reposition means putting back the ball into the box after taking it out.

#### Answers:

1. Probability of taking a white ball out:
    * $ P(white) = \frac{10}{30}=0.33 $
2. Probability of taking a white ball out after taking a black ball out:
    * $ P(black\cap white) = P(black) * P(white|black) = \frac{8}{30} \times \frac{10}{29}=0.184 $
3. Probability of taking a red ball out after taking a black and a red ball out:
    * $ P((black\cap red)\cap red) = P(black\cap red) * P(red|(black\cap red)) = \frac{8}{30} * \frac{12}{29} * \frac{11}{28} = 0.04334 $
4. Probability of taking a red ball out after taking a black and a red ball out with reposition:
    * $ P(black\cap red\cap red) = P(black) * P(red) * P(red) = \frac{8}{30} * \frac{12}{30} * \frac{12}{30} = 0.04267$

## Challenge 3

You are planning to go on a picnic today but the morning is cloudy. You hate rain so you don't know whether to go out or stay home! To help you make a decision, you gather the following data about rainy days:

* 50% of all rainy days start off cloudy!
* Cloudy mornings are common. About 40% of days start cloudy. 
* This month is usually dry so only 3 of 30 days (10%) tend to be rainy. 

What is the chance of rain during the day?

### Answers:

* $ P(cloudy) = 0.4 $
* $ P(rain|cloudy) = 0.5 $
* $ P(cloudy\cap rain) = P(cloudy) * P(rain|cloudy) = 0.4 * 0.5 = 0.2 $
* $ P(rain\cap this month) = 0.1 $
* The chaince of rain during the day would be: $ 0.2 * 0.1 = 0.02 $

## Challenge 4

One thousand people were asked through a telephone survey whether they thought more street lighting is needed at night or not.

Out of the 480 men that answered the survey, 324 said yes and 156 said no. On the other hand, out of the 520 women that answered, 351 said yes and 169 said no. 

We wonder if men and women have a different opinions about the street lighting matter. Is gender relevant or irrelevant to the question?

Consider the following events:
- The answer is yes, so the person that answered thinks that more street lighting is needed.
- The person who answered is a man.

We want to know if these events are independent, that is, if the fact of wanting more light depends on whether one is male or female. Are these events independent or not?

**Hint**: To clearly compare the answers by gender, it is best to place the data in a table.

In [2]:
# your code here
import pyforest

In [16]:
m_y = 324/480
m_n = 156/480
f_y = 351/520
f_n = 169/520
table = pd.DataFrame(data={'Yes': [m_y, f_y], 'No': [m_n, f_n]}, index=['Male', 'Female'])
table

<IPython.core.display.Javascript object>

Yes     No
Male    0.675  0.325
Female  0.675  0.325

1. The probability of a yes answer is: 
    * $ P(Yes) = (324+351)\div 1000=0.675$
2. The probability of a man answering is:
    * $ P(Man) = 480\div 1000=0.48$
3. The probability of a man answering yes is:
    * $ P(Yes\cap Man) = 0.675$
4. If two events are independent then the first bullet below must be true:
    * $ P(Yes\cap Man) = P(Yes)\ast P(Man)$
    * However, this is not the case since 0.675 <> 0.675*0.48